src/main/java/io/cdap/plugin/dlp/configs/DlpTransformConfig.java

```diff
@@ -41,14 +42,22 @@
    * @param widgetName  Name of the widget/field in the config that will be highlighted if an error is found
    * @param errorConfig Template ErrorConfig object to be used for correctly targetting errors in the FailureCollector
    */
-  void validate(FailureCollector collector, String widgetName, ErrorConfig errorConfig);
+  abstract void validate(FailureCollector collector, String widgetName, ErrorConfig errorConfig);
 
   /**
    * List of supported Schema.Type for this plugin, this will be checked against the input schema during the
    * validation stage
    *
    * @return List of supported types
    */
-  List<Schema.Type> getSupportedTypes();
+  abstract List<Schema.Type> getSupportedTypes();
 
+  /**
+   * Returns the set of fields that are required for this transform to operate correctly
+   *
+   * @return HashSet of field names that are required
+   */
+  HashSet<String> getRequiredFields() {
```

Comment:
If you prefer keeping it as interface, you can use the interface default method for this instead of changing it to abstract class.
Also, use `Set`.